### PR TITLE
Update to terra-functional-testing

### LINKED ISSUE #1354: Upgrade to use terra-functional testing
# Story

## Description
As a developer of terra I want to upgrade terra-framework to utilize terra-functional-testing.
## Motivation

So that we can get the latest features of the WDIO 6 testing infrastructure.

## Definition of Done

We'll know we're done when
* A PR is open to main referencing the snapshot of terra-functional-testing.
* All tests on PR are passing.

## Notes:
Work has been started on https://github.com/cerner/terra-framework/tree/terra-function-testing/packages
Packages that remain to be finished upgrading:

- [ ] terra-abstract-modal
- [] terra-date-time-picker

### PR DESCRIPTION
**Pull request by jmsv6d**
### Summary

Pulled in terra-functional-testing and made needed api changes for tests and resolved new accessibility errors

Closes #1354 

### Deployment Link

https://terra-framework-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-abstract-modal/tests/wdio/abstract-modal-spec.js
Comment (StephenEsser) [inline packages/terra-abstract-modal/tests/wdio/abstract-modal-spec.js:1]: Whoops, we'll want to add this to our eslint config (or whichever config) so this doesn't need to be specifically defined as a global in every spec

Comment (mjhenkes) [reply packages/terra-abstract-modal/tests/wdio/abstract-modal-spec.js:1]: https://github.com/cerner/terra-toolkit/blob/main/packages/eslint-config-terra/eslint.config.js#L74-L78

**Review by yuderekyu — COMMENTED** [review]
looks good once we pull in the eslint config and remove the inline `$` lint global. are we going to wait until we release functional testing, or merge this in with the tarball?

**Comment by mjhenkes** [comment]
We will wait until terra-functional testing releases, we won't merge this in with the tarball

File: packages/terra-slide-panel-manager/tests/wdio/slide-panel-manager-spec.js
Comment (mjhenkes) [inline packages/terra-slide-panel-manager/tests/wdio/slide-panel-manager-spec.js:201]: you can't chain these anymore?

**Review by mjhenkes — APPROVED** [review]

File: packages/terra-slide-panel-manager/tests/wdio/slide-panel-manager-spec.js
Comment (jmsv6d) [reply packages/terra-slide-panel-manager/tests/wdio/slide-panel-manager-spec.js:200]: Nope. Command chaining was removed in wdio v5.

**Review by benbcai — APPROVED** [review]

**Pull request merged**
